src/api_server/src/parsed_request.rs

```diff
@@ -270,6 +277,30 @@ pub(crate) mod tests {
         }
     }
 
+    #[test]
+    fn test_missing_slash() {
+        let (mut sender, receiver) = UnixStream::pair().unwrap();
+        let mut connection = HttpConnection::new(receiver);
+        sender
+            .write_all(
+                b"GET none HTTP/1.1\r\n\
+                Content-Type: text/plain\r\n\
+                Content-Length: 4\r\n\r\nbody",
+            )
+            .unwrap();
+        assert!(connection.try_read().is_ok());
+
+        let req = connection.pop_parsed_request().unwrap();
+        match ParsedRequest::try_from_request(&req) {
+            Err(Error::Generic(StatusCode::BadRequest, err_msg)) => {
+                if err_msg != "GET request cannot have a body." {
+                    panic!("GET request with body.");
+                }
+            }
+            _ => panic!("GET request with body."),
```

Comment:
nit: Is it necessary to check against GET with body as well?

Reply:
@iulianbarbu see update, please